core-kotlin-modules/core-kotlin-concurrency-4/src/test/kotlin/com/baeldung/continuation/ContinuationUnitTest.kt

```diff
@@ -0,0 +1,143 @@
+        }
+        simpleContinuation.resumeWith(Result.success(45))
+    }
+
+    @Test
+    fun `test continuation using suspendCoroutine`() = runBlocking {
+        val result = suspendCoroutine { continuation ->
+            continuation.resumeWith(Result.success("Baeldung"))
+        }
+        assertEquals("Baeldung", result)
+    }
+
+    // using resumeWith()
+    private suspend fun simulateNetworkRequestResume(url: String): String {
+        return withContext(Dispatchers.IO){
+            suspendCoroutine { continuation ->
+                val result = try {
+                    val connection = (URL(url).openConnection() as HttpURLConnection).apply {
+                        requestMethod = "GET"
+                        connectTimeout = 5000
+                        readTimeout = 5000
+                    }
+
+                    val responseCode = connection.responseCode
+                    if (responseCode == HttpURLConnection.HTTP_OK) {
+                        Result.success("$url : HTTP response code $responseCode - Ok")
+                    } else {
+                        Result.failure(Exception("$url: HTTP response code $responseCode - Failed"))
+                    }
+                } catch (e: Exception) {
+                    Result.failure(Exception("$url: ${e.message} - Failed"))
+                }
+
+                continuation.resumeWith(result)
+            }
+        }
+    }
+
+    // using resume() and resumeWithException()
+    private suspend fun simulateNetworkRequest(url: String): String {
+        return withContext(Dispatchers.IO){
+            suspendCoroutine { continuation ->
+                try {
+                    val connection = (URL(url).openConnection() as HttpURLConnection).apply {
+                        requestMethod = "GET"
+                        connectTimeout = 5000
+                        readTimeout = 5000
+                    }
+
+                    val responseCode = connection.responseCode
+                    if (responseCode == HttpURLConnection.HTTP_OK) {
+                        continuation.resume("$url : HTTP response code $responseCode - OK")
+                    } else {
+                        continuation.resumeWithException(Exception("$url: HTTP response code $responseCode - Failed"))
+                    }
+                } catch (e: Exception) {
+                    continuation.resumeWithException(Exception("$url: ${e.message} - Failed"))
+                }
+            }
+        }
+    }
+
+    @Test
+    fun `test continuation using suspendFunction network call`() = runBlocking {
+        val urls = listOf(
+            "https://example.com",
+            "https://example.com/fail",
+            "https://example.com/fail1",
+            "https://baeldung.com",
+            "https://www.baeldung.com/linux/",
+            "https://www.baeldung.com/kotlin/",
+            "https://www.baeldung.com/scala/",
+            "https://hangga.github.io"
+        )
```

Comment:
What's the reason for multiple URLs, can we just test one?

Also, this is no longer a UnitTest and more like a LiveTest if it relies on an external resource, so we may need to split out this test into a separate LiveTest file

Reply:
We should also split simulateNetworkRequestResume vs simulateNetworkRequest into two different tests